Performed minor overhaul

grantcurell opened the pull request:
- Made sure that the IP address and hostname fields are non-empty before continuing. Otherwise the entire script will run, but nothing will work.
- Added code guards to cause the script to fail out if any error occurs
- Added code guards to cause the script to fail out if it attempts to use an undeclared variable
- Added versioning. Everything was previously set to latest. This makes it so it is not possible to ensure things will work in the field because there is no way to guarantee users will download the same thing which was downloaded for testing.
- Added the ability to control ports at the top level
- Overhauled long lines to meet best practice for line length
- Added the ability to use IPTables instead of Firewalld IPTables rules
- Added the ability to selectively disable certain pieces of software. You may or may not want everything. TODO: Right now this does not affect the firewall rules
- Fixed a bug where nginx had the incorrect permissions. In my case, I was running this on photon so everything was owned by root which caused nginx to throw error 403

peasead committed:
Merge pull request #23 from capesstack/devel

Devel

peasead commented on deploy_capes.sh:
```diff
@@ -1,4 +1,98 @@
+################################
+
+#######################################
+# Delete a container if it already exists
```
I don't think I want to do this. I like the idea of understanding the environment and making good decisions based on that, but I don't know that I want to delete containers.

grantcurell replied:
PR Updated

peasead commented:
Thanks for this. I really like a lot of what you've done. The only one that I'd want to adjust would be not using `latest` (when already called out) and the removal of containers. 

I'm not sure how to make adjustments to your repo without closing your branch from your repo, but I'm open to learn.

peasead committed:
Merge pull request #26 from capesstack/devel

Devel

grantcurell committed:
Performed minor overhaul

- Made sure that the IP address and hostname fields are non-empty before continuing. Otherwise the entire script will run, but nothing will work.
- Added code guards to cause the script to fail out if any error occurs
- Added code guards to cause the script to fail out if it attempts to use an undeclared variable
- Added versioning. Everything was previously set to latest. This makes it so it is not possible to ensure things will work in the field because there is no way to guarantee users will download the same thing which was downloaded for testing.
- Added the ability to control ports at the top level
- Overhauled long lines to meet best practice for line length
- Added the ability to use IPTables instead of Firewalld IPTables rules
- Added the ability to selectively disable certain pieces of software. You may or may not want everything. TODO: Right now this does not affect the firewall rules
- Fixed a bug where nginx had the incorrect permissions. In my case, I was running this on photon so everything was owned by root which caused nginx to throw error 403

peasead commented:
Lots of good comments and ideas here.

There were so many, I couldn't do the merge as some were great, some needed some reworking, and some were different from the approach.

I used some of the concepts and made improvements to the project.
